Form/FormMapper.php

```diff
@@ -89,6 +89,11 @@ public function add($name, $type = null, array $options = array(), array $fieldD
 
         $group = $this->addFieldToCurrentGroup($label);
 
+        // Try to autodetect type
+        if ($name instanceof FormBuilderInterface && null !== $type) {
```

Comment:
Is this your way of agreeing with me, by showing us that Yodaism while bad for code readers / reviewers, is also bad for writers, b/c it confuses everyone ? :trollface: